imagedephi/rules.py

```diff
@@ -0,0 +1,94 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from enum import Enum
+from typing import TYPE_CHECKING
+
+import tifftools
+
+if TYPE_CHECKING:
+    from tifftools.tifftools import IFD, TagEntry
+
+
+@dataclass
+class TiffTagEntry:
+    tag: tifftools.TiffTag
+    entry: TagEntry
+
+
+class RuleType(Enum):
+    METADATA = "metadata"
+    IMAGE = "image"
+
+
+class RedactMethod(Enum):
+    REPLACE = "replace"
+    DELETE = "delete"
+    KEEP = "keep"
+
+
+class RuleFormat(Enum):
+    TIFF = "tiff"
+
+
+class RuleSource(Enum):
+    BASE = "base"
+    OVERRIDE = "override"
+
+
+@dataclass
+class Rule:
```

Comment:
Could we just add a `rule_source` within every `Rule` instance? I see that you want to know it later for printing / debugging purposes, so it seems easier to just be able to look it up directly from the rule.

Reply:
Feel free to disregard this if you think it makes constructing rules a lot more complicated.

Reply:
I agree, this makes a lot of sense. Rules knowing their own source could be helpful. Additionally it gets rid of an ad-hoc tuple data structure. Refactored in 2fe7513